src/App.js

```diff
@@ -13,14 +18,39 @@ function App() {
   // the "5" button, or the operator if they click one of those buttons) and then call your setter function to update state.
   // Don't forget to pass the functions (and any additional data needed) to the components as props
 
+  const [displayValue, setDisplayValue] = useState("");
+  const addNumber = (number) => {
+    setDisplayValue(displayValue => displayValue + number);
+  };
+
+  const addOperator = (operator) => {
+    if (operator === "=") {
+      setDisplayValue(displayValue => eval(displayValue));
+    } else {
+      setDisplayValue(displayValue => displayValue + " " + operator + " ");
+    }
+    };
+  
   return (
     <div className="container">
       <Logo />
       <div className="App">
         {/* STEP 4 - Render your components here and be sure to properly import/export all files */}
+      <Display number={displayValue} />
+
+      <div className="left">
+      <Specials />
+      <Numbers addNumber={addNumber} />      
+      </div>
+      
+      <div className= "ops">
+      <Operators addOperator={addOperator} />
+      </div>
+      
+      
       </div>
     </div>
   );
 }
 
-export default App;
+export default App;
```

Comment:
This is awesome, you got the stretch and the code is easy to follow! I think the only thing you need is a function to clear the display, that shouldn't be too hard, think through the logic like this: "if (specialbutton === "C") => setDisplay("0"); Of course you have to prop drill and maybe change the variable to make it work.